examples/08_Advanced_Usage/TorchScript_Exact_Models.ipynb

```diff
@@ -0,0 +1,190 @@
+    "        super(ExactGPModel, self).__init__(train_x, train_y, likelihood)\n",
+    "        self.mean_module = gpytorch.means.ConstantMean()\n",
+    "        self.covar_module = gpytorch.kernels.ScaleKernel(gpytorch.kernels.RBFKernel())\n",
+    "    \n",
+    "    def forward(self, x):\n",
+    "        mean_x = self.mean_module(x)\n",
+    "        covar_x = self.covar_module(x)\n",
+    "        return gpytorch.distributions.MultivariateNormal(mean_x, covar_x)\n",
+    "\n",
+    "# initialize likelihood and model\n",
+    "likelihood = gpytorch.likelihoods.GaussianLikelihood()\n",
+    "model = ExactGPModel(train_x, train_y, likelihood)\n",
+    "\n",
+    "# this is for running the notebook in our testing framework\n",
+    "import os\n",
+    "smoke_test = ('CI' in os.environ)\n",
+    "training_iter = 2 if smoke_test else 50\n",
+    "\n",
+    "\n",
+    "# Find optimal model hyperparameters\n",
+    "model.train()\n",
+    "likelihood.train()\n",
+    "\n",
+    "# Use the adam optimizer\n",
+    "optimizer = torch.optim.Adam([\n",
+    "    {'params': model.parameters()},  # Includes GaussianLikelihood parameters\n",
+    "], lr=0.1)\n",
+    "\n",
+    "# \"Loss\" for GPs - the marginal log likelihood\n",
+    "mll = gpytorch.mlls.ExactMarginalLogLikelihood(likelihood, model)\n",
+    "\n",
+    "for i in range(training_iter):\n",
+    "    # Zero gradients from previous iteration\n",
+    "    optimizer.zero_grad()\n",
+    "    # Output from model\n",
+    "    output = model(train_x)\n",
+    "    # Calc loss and backprop gradients\n",
+    "    loss = -mll(output, train_y)\n",
+    "    loss.backward()\n",
+    "    optimizer.step()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Trace the Model\n",
+    "\n",
+    "In the next cell, we trace our GP model. To overcome the fact that we can't trace Modules that return Distributions, we write a wrapper Module that unpacks the GP output in to a mean and variance.\n",
+    "\n",
+    "Then, before calling `torch.jit.trace` we first call the model on `test_x`. This step is **required**, as it does some precomputation using torch functionality that cannot be traced."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class MeanVarModelWrapper(torch.nn.Module):\n",
+    "    def __init__(self, gp):\n",
+    "        super().__init__()\n",
+    "        self.gp = gp\n",
+    "    \n",
+    "    def forward(self, x):\n",
+    "        output_dist = self.gp(x)\n",
+    "        return output_dist.mean, output_dist.variance\n",
+    "\n",
+    "with torch.no_grad(), gpytorch.settings.fast_pred_var():\n",
+    "    model.eval()\n",
+    "    test_x = torch.linspace(0, 1, 51)\n",
+    "    pred = model(test_x)  # Do precomputation\n",
+    "    traced_model = torch.jit.trace(MeanVarModelWrapper(model), test_x, check_trace=False)"
```

Comment:
Have you guys tried saving the traced model by "traced_model.save('model_state.pt')“ by any chance? Saving will cause error on my side due to "RuntimeError: Could not export Python function call 'RBFCovariance'. Remove calls to Python functions before export." 
We have the need for saving the .pt model and use it for C++ online inferences. Thank you and look forward to your reply!

Reply:
Sorry for the delay finishing up this PR -- things are just getting a little busy as we approach the ICML deadline.

I'll take a look. What we may have to do is somehow avoid using the custom Functions for RBF and Matern when tracing.